external/fv3fit/fv3fit/keras/_models/shared/pure_keras.py

```diff
@@ -115,8 +116,96 @@ def predict(self, X: xr.Dataset) -> xr.Dataset:
                         attrs={"units": X_stacked["specific_humidity"].units + " / s"},
                     ),
                 }
+            ).unstack(SAMPLE_DIM_NAME)
+        return match_prediction_to_input_coords(X, return_ds)
+
+    def dump(self, path: str) -> None:
+        with put_dir(path) as path:
+            if self.model is not None:
+                model_filename = os.path.join(path, self._MODEL_FILENAME)
+                self.model.save(model_filename)
+            with open(os.path.join(path, self._CONFIG_FILENAME), "w") as f:
+                f.write(
+                    yaml.dump(
+                        {
+                            "input_variables": self.input_variables,
+                            "output_variables": self.output_variables,
+                            "output_metadata": self._output_metadata,
+                        }
+                    )
+                )
+
+
+@io.register("all-keras-no-stack")
+class PureKerasNoStackModel(Predictor):
+    """Model which uses Keras for packing and normalization"""
+
+    _MODEL_FILENAME = "model.tf"
+    _CONFIG_FILENAME = "config.yaml"
+
+    def __init__(
+        self,
+        input_variables: Sequence[Hashable],
+        output_variables: Sequence[Hashable],
+        output_metadata: Iterable[Dict[str, Any]],
+        model: tf.keras.Model,
+        loss=None,
```

Comment:
Does the Predictor need the loss? It doesn't look like it's dumped or loaded.

Reply:
This class is now deleted.